telegram-desktop: update to 4.5.2

RedL0tus opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

Upgrade telegram-desktop to 4.5.2
<!-- Please input topic description here. -->

Package(s) Affected
-------------------

abseil-cpp: update to 20220623.1
grpc: bump rel for abseil-cpp 20220623.1
bear: bump rel for abseil-cpp 20220623.1
mumble: update to 1.4.287
libsigc++-3.0: new, 3.4.0
glibmm-2.68: new, 2.72.1
telegram-desktop: update to 4.5.2

<!-- Please list all package(s) affected by this topic here. -->

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

Build Order
-----------

`abseil-cpp grpc bear mumble libsigc++-3.0 glibmm-2.68 telegram-desktop`

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

RedL0tus committed:
abseil-cpp: update to 20220623.1

RedL0tus committed:
grpc: bump rel for abseil-cpp 20220623.1

RedL0tus committed:
bear: bump rel for abseil-cpp 20220623.1

RedL0tus committed:
mumble: update to 1.4.287 for abseil-cpp 20220623.1

    - FTBFS while attempting to bump rel for abseil-cpp 20220623.1,
      however, this new version werks.
    - Remove 0001-BUILD-cmake-Find-and-link-Poco-XML.patch, as it
      is already upstreamed (39e2030).
    - Explicitly set ABTYPE to cmakeninja.

RedL0tus committed:
libsigc++-3.0: new, 3.4.0

... for glibmm-2.68.

RedL0tus committed:
glibmm-2.68: new, 2.72.1

    - For telegram-desktop 4.5.2.
    - Latest usable version of glibmm without a newer glib.

RedL0tus committed:
telegram-desktop: update to 4.5.2

    - Now glibmm is required on AMD64 as well.
    - Rebase all patches to 4.5.2.

Petition to add GNOME to the list of unapproved desktop
environments.

MingcongBai commented on extra-gnome/glibmm-2.68/autobuild/defines:
```diff
@@ -0,0 +1,23 @@
+BUILDDEP__LOONGSON2F="${BUILDDEP__RETRO}"
+BUILDDEP__POWERPC="${BUILDDEP__RETRO}"
+BUILDDEP__PPC64="${BUILDDEP__RETRO}"
+PKGDES="C++ interface for GLib"
```
Add "(2.68 ABI)" or something like that to distinguish.

MingcongBai commented on extra-libs/libsigc++-3.0/autobuild/defines:
```diff
@@ -0,0 +1,23 @@
+PKGNAME=libsigc++-3.0
+PKGDES="A typesafe callback system for standard C++ version 3.0"
```
Move "version 3.0" into parentheses.

RedL0tus committed:
telegram-desktop: disable LTO on riscv64

    - LTO build fails on riscv64 with the following message:
      LTO_symtab_tags out of range: Range is 0 to 5, value is 237.
      Disable it for now.
